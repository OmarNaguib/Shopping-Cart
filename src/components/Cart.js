import LabelRow from "./LabelRow";

export default function Cart({ removeFromCart, increment, decrement, items }) {
  const checkItems = items
    .filter((item) => item.inCart)
    .map((item, index) => {
      return (
        <div className="checkout-item">
          <div className="product">
            <img src={item.src} alt="" />
            <div className="name">{item.name}</div>
          </div>
          <div className="price">£{item.price}</div>
          <div className="quantity">
            <div className="number">{item.inCart}</div>
            <div className="buttons">
              <button onClick={increment.bind(null, item.id)}>+</button>
              <button onClick={decrement.bind(null, item.id)}>-</button>
            </div>
          </div>
          <div className="subtotal">£{item.price * item.inCart}</div>
          <button onClick={removeFromCart.bind(null, item.id)}>delete</button>
        </div>
      );
    });
  return (
    <div className="cart">
      <div className="checkout-card">
        <h1>Cart</h1>
        <LabelRow></LabelRow>
        {checkItems}
        <button className="checkout">Checkout</button>
      </div>
    </div>
  );
}
