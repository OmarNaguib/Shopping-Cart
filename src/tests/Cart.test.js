import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Cart from "../components/Cart";

const items = [
  {
    src: "/static/media/1table_JOOLA J15 INSIDE TABLE TENNIS TABLE_599.00.dd9ef9f4bbc1398f6cc7.jpg",
    name: "JOOLA J15 INSIDE TABLE TENNIS TABLE",
    price: "599.00",
    inCart: 0,
  },
];

describe("Cart page", () => {
  it("renders correctly with one item", () => {
    const { container } = render(
      <Cart
        items={items}
        removeFromCart={jest.fn()}
        increment={jest.fn()}
        decrement={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
