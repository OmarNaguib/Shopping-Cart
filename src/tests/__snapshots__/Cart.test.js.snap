// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart page renders correctly with one item 1`] = `
<div>
  <div
    class="cart"
  >
    <div
      class="checkout-card"
    >
      <h1>
        Cart
      </h1>
      <div
        class="label-row"
      >
        <div
          class="product"
        >
          Product
        </div>
        <div
          class="price"
        >
          Price
        </div>
        <div
          class="quantity"
        >
          Quantity
        </div>
        <div
          class="subtotal"
        >
          subtotal
        </div>
      </div>
      <div
        class="check"
      >
        <div>
          <span>
            Grand total : 
          </span>
          <span
            class="amount"
            data-testid="grand-total"
          >
            Â£
            0
          </span>
        </div>
        <button
          class="checkout"
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;
